apps/astarte_realm_management/lib/astarte_realm_management/queries.ex

```diff
@@ -1027,6 +1027,136 @@ defmodule Astarte.RealmManagement.Queries do
     end
   end
 
+  def get_interfaces_details(client) do
+    interfaces_details_query = """
+    SELECT * FROM interfaces
+    """
+
+    query =
+      DatabaseQuery.new()
+      |> DatabaseQuery.statement(interfaces_details_query)
+      |> DatabaseQuery.consistency(:quorum)
+
+    with {:ok, interfaces_result} <- DatabaseQuery.call(client, query) do
+      interfaces =
+        Enum.map(interfaces_result, fn interface_row ->
+          %{
+            name: name,
+            major_version: major_version,
+            minor_version: minor_version,
+            interface_id: interface_id,
+            type: type,
+            ownership: ownership,
+            aggregation: aggregation,
+            description: description,
+            doc: doc
+          } = Enum.into(interface_row, %{})
+
+          %InterfaceDocument{
+            name: name,
+            major_version: major_version,
+            minor_version: minor_version,
+            interface_id: interface_id,
+            type: InterfaceType.from_int(type),
+            ownership: Ownership.from_int(ownership),
+            aggregation: Aggregation.from_int(aggregation),
+            mappings: %{},
+            description: description,
+            doc: doc
```

Comment:
Here you could probably use the [`InterfaceDescriptor.from_db_result/1`](https://hexdocs.pm/astarte_core/Astarte.Core.InterfaceDescriptor.html#from_db_result/1) function from `astarte_core`